### PR TITLE
Intra-table federation query on duckdb accelerated table

### PR DESCRIPTION
**Pull request by y-f-u**
## 🗣 Description

Intra table federation for accelerated table.

## 🔨 Related Issues

part of #2182 

image

## Concerns

- when there is a use_source action on zero result, the federation will be disabled to use original logic. This can be enhanced introducing a virtual execution to handle this case.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -116,3 +116,6 @@ arrow-odbc = { git = "https://github.com/spiceai/arrow-odbc.git", rev = "24ecbdf
 
 # Tracking Issue: https://github.com/allan2/dotenvy/issues/113
 dotenvy = { git = "https://github.com/spiceai/dotenvy.git", rev = "e5cef1871b08003198949dfe2da988633eaad78f" }
+
+[patch."https://github.com/datafusion-contrib/datafusion-table-providers.git"]
+datafusion-table-providers = { path = "../../datafusion-contrib/datafusion-table-providers" }
```
Comment (y-f-u) [inline Cargo.toml]: organising now

**Comment by y-f-u** [comment]
tests are failing, working on a fix.

File: crates/data_components/src/poly.rs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+};
+use datafusion_federation::{FederatedTableProviderAdaptor, FederationProvider};
+
+use crate::delete::{get_deletion_provider, DeletionTableProvider};
+
+#[derive(Clone)]
+pub struct PolyTableProvider {
+    write: Arc<dyn TableProvider>,
+    delete: Arc<dyn TableProvider>,
+    fed: Arc<dyn TableProvider>,
```
Comment (phillipleblanc) [inline crates/data_components/src/poly.rs]: Why not just store the values with their concrete types?

Comment (y-f-u) [reply crates/data_components/src/poly.rs]: This is meant to be handling all types of accelerator. The only thing can be concrete is delete which can use the adaptor. Will change it so.

**Comment by y-f-u** [comment]
The test is broken for a known issue from sqlite. Which @peasee is looking at for a fix.

File: crates/runtime/src/datafusion.rs [hunk 3]
```diff
@@ -818,9 +826,35 @@ impl DataFusion {
                     table_name: dataset_name.to_string(),
                 })?;
         }
+
         Ok(())
     }
 
+    pub async fn get_accelerated_table_provider(
```
Comment (y-f-u) [inline crates/runtime/src/datafusion.rs:833]: this method doesn't return a &AcceleratedTable due to lifetime issue that `table` can't be used in returned reference.

Let me know if any better way to deal with this.

**Review by phillipleblanc — DISMISSED** [review]

**Review by phillipleblanc — APPROVED** [review]

**Pull request merged**